tests/pyfunc/test_chat_model.py

```diff
@@ -380,3 +380,15 @@ def test_chat_model_predict_stream(tmp_path):
 
     response = next(loaded_model.predict_stream({"messages": messages}))
     assert response["choices"][0]["message"]["content"] == json.dumps(messages)
+
+
+# test that users cannot overwrite the 'object' field in ChatResponse
+def test_chat_model_response_cannot_overwrite_object():
```

Comment:
could we also test if someone tries to set the property when initializing ChatResponse? e.g. `ChatResponse(..., object="something")`. i think this case should also throw (it seems like the message will be generic if this happens, but that might be hard to prevent)

Reply:
good catch

Reply:
from our conversation offline though i'm not sure if maybe this is something important to support or not 🤔. i feel like the only real use-case is if people are trying to dump openAI responses directly to ChatResponse, like
```python
response = ...  # call openAI
return ChatResponse(**response)
```

but i don't know if this is common or not

Reply:
I actually reverted my solution in order to support this. I did see this usage at least in our own tests and codebase. don't have much context beyond that

Reply:
Yes I agree, it is P0 requirement that users can pass the raw OpenAI response dictionary to the constructor without needing to sanitize it.